contrib/signet/getcoins.py

```diff
@@ -120,22 +122,34 @@ def bitcoin_cli(rpc_command_and_params):
 if args.captcha != '': # Retrieve a captcha
     try:
         res = session.get(args.captcha)
-    except:
-        print('Unexpected error when contacting faucet:', sys.exc_info()[0])
-        exit(1)
+        res.raise_for_status()
+    except requests.exceptions.RequestException as e:
+        raise SystemExit(f"Unexpected error when contacting faucet: {e}")
+
+    # Size limitation
+    svg = xml.etree.ElementTree.fromstring(res.content)
+    if svg.attrib.get('width') != '150' or svg.attrib.get('height') != '50':
+        raise SystemExit("Captcha size doesn't match expected dimensions 150x50")
```

Comment:
Tested this part of code by using different url for captcha in getcoins.py:

```python
DEFAULT_GLOBAL_CAPTCHA = 'https://prayank23.github.io/svg-test/size.svg'
```
and this svg file:

```xml
<svg xmlns="http://www.w3.org/2000/svg" height="300" width="100">
  <text x="0" y="15">Size test</text>
</svg>
```

Works as expected:

```
$ ./getcoins.py 

Captcha size doesn't match expected dimensions 150x50
```